### DISCUSSION TITLE
Remapping copilot#Accept works, but it still accepting with TAB key

### DISCUSSION THREAD

**Discussion by BubuDavid** [community/community#118483] [Copilot Conversations] [answered]
### Select Topic Area

Bug

### Body

# Setup
Hi there 👋
```bash
NVIM v0.9.4
Build type: Release
Linux VM-Processor 6.5.0-1017-azure #17~22.04.1-Ubuntu SMP but also in WSL2 with Ubuntu 22.04.
I am using Lazy as my plugin manager for nvim.
```
You can see my exact configuration [here](https://github.com/bubuDavid/nvim/)

# Bug
I have the following configuration from Lazy for copilot, you can see my configuration over there 👆 if you want more details (or ask me if you prefer):
```lua
return {
	"github/copilot.vim",
	config = function()
		vim.keymap.set("i", "<C-j>", 'copilot#Accept("\\<CR>")', {
			expr = true,
			replace_keycodes = false,
		})
		vim.g.copilot_no_tab_map = true
		vim.g.copilot_assume_mapped = true
	end,
}
```
The problem is, when a suggestion appear, when I press **Tab** key, it accept it, I do not know what to do and I try reading a bunch of forums and answers, if anyone could help me it  would be amazing.

**Comment by tpope** [comment]
I suspect what's happening is the `config` function is running after the map is already created. Try moving that `vim.g.copilot_no_tab_map = true` to the top level and see if that clears it up.

**Reply by BubuDavid** [reply]
I already tried that, failed, still accepting the suggestion after tabing :c

**Reply by tpope** [reply, marked as answer]
Check `:verbose imap <Tab>` to confirm that `plugin/copilot.vim` is indeed defining the map. Make sure it's the right path (i.e., that you don't have 2 versions of the plugin installed.)

Check `:scriptnames` to see what order runtime files are loading. Make sure the file that sets the variable loads before any file from Copilot.vim.

**Reply by BubuDavid** [reply]
YOU'R A GENIUS! I had to run those commands before the copilot initialization, thanks that worked amazingly, I put the commands:
```
vim.g.copilot_no_tab_map = true
vim.g.copilot_assume_mapped = true
```
In my `keymaps.lua` file instead of the copilot configuration file, cool...
It is kind of weird that I need to do the copilot configurations outside copilot file, but I can live with that. **For anyone that may have been passing for the same problem as me, check my nvim configuration repo in the description of the bug and see specially the `lua/buberto/core/keymaps.lua` and the `lua/buberto/plugins/copilot.lua` files, that is all you need**

Thanks! 🤗
